File: InPlan.Web/Modules/Parametros/Ambientes/AmbientesForm.cs
using Serenity.ComponentModel;

namespace InPlan.Parametros.Forms;

[FormScript("Parametros.Ambientes")]
[BasedOnRow(typeof(AmbientesRow), CheckNames = true)]
public class AmbientesForm
{
    [HalfWidth]
    public int IdEmpresa { get; set; }
    [HalfWidth]
    public int IdArea { get; set; }
    public string Descripcion { get; set; }
    [TextAreaEditor(Rows = 5)]
    public string Observaciones { get; set; }
    [HideOnInsert]
    public bool Baja { get; set; }
}

File: InPlan.Web/Modules/ServerTypes/Parametros/AmbientesForm.ts
import { IntegerEditor, StringEditor, BooleanEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface AmbientesForm {
    IdEmpresa: IntegerEditor;
    IdArea: IntegerEditor;
    Descripcion: StringEditor;
    Observaciones: StringEditor;
    Baja: BooleanEditor;
}

export class AmbientesForm extends PrefixedContext {
    static formKey = 'Parametros.Ambientes';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!AmbientesForm.init)  {
            AmbientesForm.init = true;

            var w0 = IntegerEditor;
            var w1 = StringEditor;
            var w2 = BooleanEditor;

            initFormType(AmbientesForm, [
                'IdEmpresa', w0,
                'IdArea', w0,
                'Descripcion', w1,
                'Observaciones', w1,
                'Baja', w2
            ]);
        }
    }
}